src/front/app/components/Challenges/ChallengeList.js

```diff
@@ -1,7 +1,47 @@
 import React from 'react';
+import { connect } from 'react-redux';
+import Challenge from './Challenge';
+import TopChallenge from './TopChallenge';
 
-export default () => (
-    <div>
-        ChallengeList
-    </div>
-);
+let ChallengeList = ({ challenges, dispatch }) => {
+    if (!challenges.length) {
+        dispatch(getInitialChallenges());
+    }
+    const first = challenges[0];
+    const others = challenges.slice(1);
+
+    return (
+        <ul className="challenges__list">
+            <TopChallenge { ...first } />
+            { others.map((challenge, index) => (
+                    <Challenge key={ index } { ...challenge } />
+            ))}
+        </ul>
+    );
+};
+
+ChallengeList.propTypes = {
+    challenges: React.PropTypes.array,
+    dispatch: React.PropTypes.func
+};
+
+const mapStateToProps = ({ challenges }) => ({ challenges });
+
+function getInitialChallenges() {
```

Comment:
I meant the action function should be in the separate module (grouped with other actions by related types or so) and be imported to the component module - it would allow its separate testing, make it reusable, and would not create mixing of concerns in the view component (which should know nothing about communication with the server)